win32/src/winapi/user32/window.rs

```diff
@@ -869,7 +869,20 @@ pub fn MoveWindow(
 
 #[win32_derive::dllexport]
 pub fn GetClientRect(machine: &mut Machine, hWnd: HWND, lpRect: Option<&mut RECT>) -> bool {
-    let window = machine.state.user32.windows.get_mut(hWnd).unwrap();
+    let window = machine.state.user32.windows.get(hWnd).unwrap();
+    let rect = lpRect.unwrap();
+    *rect = RECT {
+        left: 0,
+        top: 0,
+        right: window.width as i32,
+        bottom: window.height as i32,
+    };
+    true
+}
+
+#[win32_derive::dllexport]
+pub fn GetWindowRect(machine: &mut Machine, hWnd: HWND, lpRect: Option<&mut RECT>) -> bool {
```

Comment:
This is subtle (and I should have documented it better, sorry!).  The 'client area' is the part without the title bar / borders, while the 'window rect' includes the title bar and borders.  On most platforms when you size a window you size it by the client area dimensions, so the underlying `window.width` is in terms of client area, which is why GetClientRect can return window.width directly.

But for GetWindowRect, you need to offset it by the window's decorations dimensions.

tldr see the function `window_size` in this file, you need to pass the rect through that :)

Reply:
(Also for the style/menu bits, if they're not convenient right now feel free to just add TODO comments)

Reply:
Makes sense! I thought that it was a bit weird that the implementations where the same 😅

Reply:
https://stackoverflow.com/a/7561187/148072

It seems like `GetWindowRect` should also return screen coordinates. Right now we don't get x/y position from the host, so I'm setting them to 0,0 for the time being...

Reply:
Seems to be working!

<img width="1250" alt="Screenshot 2024-09-14 at 23 54 02" src="https://github.com/user-attachments/assets/d7513b22-652a-4296-ad4e-b605c81336b0">